StonehearthEditor/Recipes/RecipesView.cs

```diff
@@ -0,0 +1,701 @@
+
+                // Add materials
+                strings.AddRange(mMaterialImages.Keys);
+
+                ModuleDataManager mdm = ModuleDataManager.GetInstance();
+                Dictionary<string, JsonFileData> itemJsons = mdm.GetJsonsByTerm("entity_data.stonehearth:net_worth");
+                foreach (KeyValuePair<string, JsonFileData> kv in itemJsons)
+                {
+                    string modName = mdm.GetModNameFromAlias(kv.Key);
+                    bool shouldIncludeMod = mBaseModsOnly ? (modName == "stonehearth" || modName == "rayyas_children") : true;
+                    if (shouldIncludeMod)
+                    {
+                        strings.Add(kv.Key);
+                    }
+                }
+            }
+
+            return strings;
+        }
+
+        private string GetDisplayName(JsonFileData jsonFileData)
+        {
+            JsonFileData mixinsFile = jsonFileData.CreateFileWithMixinsApplied();
+            JToken displayName = mixinsFile.Json.SelectToken("entity_data.stonehearth:catalog.display_name");
+            return displayName.ToString();
+        }
+
+        public Image GetIcon(JsonFileData jsonFileData)
+        {
+            foreach (KeyValuePair<string, FileData> kv in jsonFileData.LinkedFileData)
+            {
+                FileData fileData = kv.Value;
+                string path = "";
+                if (fileData is JsonFileData)
+                {
+                    path = (fileData as JsonFileData).GetImageForFile();
+                }
+                else if (fileData is ImageFileData)
+                {
+                    path = fileData.Path;
+                }
+
+                if (path != "" && System.IO.File.Exists(path))
+                {
+                    return ThumbnailCache.GetThumbnail(path);
+                }
+            }
+
+            return null;
+        }
+
+        public Image GetIcon(string material)
+        {
+            string path;
+            mMaterialImages.TryGetValue(material, out path);
+            if (path != null)
+            {
+                return ThumbnailCache.GetThumbnail(path);
+            }
+
+            return null;
+        }
+
+        private void searchBox_Filter(object sender, EventArgs e)
+        {
+            string searchTerm = searchBox.Text.ToString();
+            string colFilter = filterCmbx.Text;
+            if (searchTerm != "")
+            {
+                searchBox.BackColor = Color.Gold;
+            }
+            else
+            {
+                searchBox.BackColor = Color.White;
+            }
+
+            StringBuilder sb = new StringBuilder();
+            // Filter by all columns or all ingredient columns
+            if (colFilter == kAllCol || colFilter.Contains(IngredientColumnGroup.kIngr))
+            {
+                Regex matchIng = new Regex(IngredientColumnGroup.kIngr + "\\d");
+                for (int i = 0; i < recipesGridView.Columns.Count; i++)
```

Comment:
Foreach?

Reply:
i use the index to check whether we are not the last column in the grid